packages/hebao_v1/contracts/modules/transfers/DappTransfers.sol

```diff
@@ -0,0 +1,133 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.6.6;
+pragma experimental ABIEncoderV2;
+
+import "../../lib/AddressSet.sol";
+import "../../lib/ERC20.sol";
+
+import "./TransferModule.sol";
+
+
+/// @title DappTransfers
+contract DappTransfers is TransferModule
```

Comment:
If `executeTransfers` is removed (I'm still not convinced! ;) then only being able to call a single function (+ optional approve) in a meta tx is probably a serious limitation.

Reply:
If we truly want to support atomic batch calls to an external dapp, we can write a custom module to do that, it's probably easier and safer.